Feat/84 카페 수정

gaeul79 opened the pull request:
## 🔘 담당 파트
- 카페 수정

## 🔎 작업 상세 내용
- 카페에 대한 기본 정보를 수정한다.
- 카페 이미지, 입장료, 방, 가격정책은 카페 수정 api에서 수정하지 않는다.

## 🔧 앞으로의 과제
-  카페 등록(admin)
- 카페 리스트 조회(관리자 검색한)
- 카페 리스트 조회(사장님이 등록한)
- 카페 삭제(ADMIN)
- 카페 삭제(OWNER)
- 검색시 지역명을 불러오도록 하는 기능 구현 필요

## ➕ 이슈 링크
- #84

Commit:
-feat : 신고 상태 변경 기능 구현

Commit:
-feat : 리뷰 컨트롤러 수정

Commit:
-feat : 테스트 코드 작성

Commit:
-feat : 룸 생성 기능 구현 완료

Commit:
-feat : 룸 조회 기능 구현 완료

Commit:
-feat : 룸 수정 기능 구현 완료

gaeul79 committed:
feat: 카페 수정 기능 구현
- 카페와 카페 이미지를 수정하는 기능 구현
- 카페 수정 테스트 코드 작성

gaeul79 committed:
fix: 카페 수정 버그 수정
- db에서 카페 이미지가 삭제되지 않는 현상 수정
- cafe entity orphanRemoval = true옵션 추가
- BookmarkServiceTest Cafe 생성부분 builder패턴으로 변경

gaeul79 committed:
fix: 문법 오류 수정

gaeul79 reviewed (commented):
어라........ ?

gaeul79 committed:
fix: 문법 오류 수정

shushuaka reviewed (approved):
뭣이 이리 많이 했대요 wow
╰(*°▽°*)╯

gaeul79 committed:
refactor: 카페 수정 이미지 관련 코드 리팩토링
- DB는 트랜젝션이 걸리므로 튜터님 피드백 반영
- DB등록이 정상적으로 됐을시에 파일 업로드 진행
- FileUtil 이미지 로직이 들어가는 쪽은 공통으로 사용할 수 있도록 보완
- 추루 S3에 비슷하게 리펙토링 Config로 빼는것도 고려
- 이미지 관련 에러코드 수정

gaeul79 committed:
Merge branch 'dev' into feat/84-카페-수정
